src/components/button/Button.tsx

```diff
@@ -0,0 +1,91 @@
+import styled from "@emotion/styled";
+import Image from "next/image";
+import { useState, ButtonHTMLAttributes } from "react";
+import { getButtonStyles, getWidthStyles, setLoadingStyles } from "components/button/styles";
+
+export interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: "primary" | "secondary" | "text" | "icon" | "iconWithText"; // color types
```

Comment:
오.. variant 가 다양하군요!

이건 제 디자인 미스인 것 같습니다. ㅜ.ㅜ

모든 종류의 버튼에는 아이콘이 들어갈 수 있는 것으로 생각하고 디자인 하긴 했습니다 ㅎ

그렇다면  `icon` 을 옵셔널하게  props 로 받아 icon 값이 주어지면  14*14 사이즈로 글씨 옆에 보여주고, 주어지지 않으면 아이콘을 노출하지 않는 것은 어떨까요~?

```suggestion
  variant?: "primary" | "secondary" | "text" | "icon" ; // color types
  icon?: React.ReactNode;
```

Reply:
근데 icon과 iconWithText의 구분을 어떻게 해야 할까요....? children 여부로 하면 될까요?

Reply:
@hoon5083 음 ... 그것도 좀 애매하네요. 차라리 `icon` 타입도 없애고, 

* `icon` 있음, children 없음 => 아이콘만 있는 정방형버튼
* `icon` 있음, children 있음 => (기존의) `iconWithText` 

이렇게 되는 것은 어떨까요?

Reply:
넵넵 좋습니다! @jaychang99 님 말씀대로 해보겠습니다.

Reply:
이부분 해결된 것 같아 resolve 합니다~수고 하셨습니다~